pipeline/metadata/ip_metadata_chooser.py

```diff
@@ -0,0 +1,89 @@
+"""A module which contains a number of sources of ip metadata and chooses how to add them to ips."""
+import datetime
+import logging
+
+from pipeline.metadata import caida_ip_metadata
+from pipeline.metadata import dbip
+from pipeline.metadata.flatten import Row
+
+
+# pylint: disable=too-many-instance-attributes
+class IpMetadataChooser():
+  """Business logic for selecting IP metadata values from a number of sources"""
+
+  def __init__(self, caida_db: caida_ip_metadata.CaidaIpMetadata,
+               dbip_db: dbip.DbipMetadata) -> None:
+    """Store all metadata sources for future querying
+
+    Args:
+      caida: CAIDA database
+      dbip: dbip database
+    """
+    self.caida = caida_db
+    self.dbip = dbip_db
+
+  def get_metadata(self, ip: str) -> Row:
+    """Pick which metadata values to return for an IP from our sources."""
+    try:
+      (netblock, asn, as_name, as_full_name, as_type,
+       country) = self.caida.lookup(ip)
+      metadata_values = {
+          'netblock': netblock,
+          'asn': asn,
+          'as_name': as_name,
+          'as_full_name': as_full_name,
+          'as_class': as_type,
+          'country': country,
+      }
+
+      (org, dbip_asn) = self.dbip.lookup(ip)
```

Comment:
It may be a little clearer now how you could simply pass a DBIP `mmdb_reader` in the constructor and not need the `DbipMetadata` library. But this looks good.

Reply:
I'd like to do something smarter with DbipMetadata in the future (maybe use multiple tables, we'll see) So I'd like to keep the separation.